### PR TITLE
Bump ProtoData and Validation

### PR DESCRIPTION
**Pull request by alexander-yevsyukov**
This PR updates dependences on ProtoData, Validation, and others from the recent `config`.

Other notable changes:
  * Fat JAR of `mc-java-protoc` was trimmed to avoid carrying really unnecessary files.
  * Introduced `ApiOption` enum.
  * Added proto types for ProtoData-based implementation of source code annotations.

### CODE REVIEW

**Comment by alexander-yevsyukov** [comment]
@armiol, there's nothing interesting in this PR. It's just adaptation to newer API form ProtoData and drafts of entities for new annotations. So, I'm going to merge it, reducing the amount of really interesting stuff, when it comes to new annotations.

File: mc-java-annotation/src/main/kotlin/io/spine/tools/mc/annotation/ApiOption.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import io.spine.protodata.option
+import com.google.protobuf.Any as ProtoAny
+
+/**
+ * This object simulates an enum of options, which are used to mark API elements.
+ *
+ * We use this arrangement to allow lazy instantiation of elements, also avoiding
+ * boilerplate code of enum declaration in this particular case.
+ */
+internal enum class ApiOption(optionName: String) {
```
Comment (armiol) [inline mc-java-annotation/src/main/kotlin/io/spine/tools/mc/annotation/ApiOption.kt:41]: As far as I understand, the list of options in this type describes only file-level options.

Maybe, we should reflect this idea in either enum name, or the documentation. Or both.

Please consider.

File: mc-java-annotation/src/main/proto/spine/tools/mc/annotation/annotated_types.proto [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+//
+// API level options can be also applied to fields of a message. For example, `experimental_field`
+// makes only the field experimental, while the message itself remains stable.
+//
+// API level options are used to produce corresponding annotations in the generated code.
+//
+
+// A type which has been annotated with one or several API level options.
+//
+// The type may be a message, an enum, or a service.
```
Comment (armiol) [inline mc-java-annotation/src/main/proto/spine/tools/mc/annotation/annotated_types.proto:66]: > ... or a service.

In which case, this type can hardly be named `...Message`. I would say, it's something like `AnnotatedType`, or `AnnotatedDeclaration`.

File: mc-java-annotation/src/main/proto/spine/tools/mc/annotation/annotated_types.proto [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+    // The field with the options applied.
+    protodata.Field field = 1 [(required) = true];
+
+    // One or more applied options.
+    repeated protodata.Option option = 2 [(required) = true];
+}
+
+// The API level options applied to an enum.
+message AnnotatedEnum {
```
Comment (armiol) [inline mc-java-annotation/src/main/proto/spine/tools/mc/annotation/annotated_types.proto:100]: Given `AnnotatedMessage` describes all of them:
> The type may be a message, an enum, or a service.

I don't understand the idea of having `AnnotatedEnum` and `AnnotatedService` (below).

File: mc-java-validation/src/main/java/io/spine/tools/mc/java/validation/gen/ConstraintCode.java [hunk 4]
```diff
@@ -126,14 +126,14 @@ private static CodeBlock evaluate(CodeBlock declarations,
      *
      * @return new builder
      */
-    public static Builder forField(FieldDeclaration field) {
+    static Builder forField(FieldDeclaration field) {
         return new Builder(field);
```
Comment (armiol) [inline mc-java-validation/src/main/java/io/spine/tools/mc/java/validation/gen/ConstraintCode.java:130]: I would check the input for `null`.

**Review by armiol — CHANGES_REQUESTED** [review]
@alexander-yevsyukov please see my comments.

**Review by armiol — APPROVED** [review]
LGTM.

**Pull request merged**